packages-ts/starknet-gauntlet/src/dependencies/index.ts

```diff
@@ -6,15 +6,23 @@ export interface Env {
   providerUrl: string
   pk?: string
   account?: string
-  [key: string]: string // Custom env
+  withLedger?: boolean
+  ledgerPath?: string
+  multisig?: string
+  [key: string]: string | boolean // Custom env
 }
 
 export interface Dependencies {
   logger: typeof logger
   prompt: typeof prompt
   makeProvider: (url: string) => IStarknetProvider
-  makeWallet: (pk: string, account?: string) => IStarknetWallet
-  makeEnv: (flags: Record<string, string>) => Env
+  makeWallet: (
```

Comment:
It makes sense. Please have another look, i think it looks better now. 
I'm not sure about separate `starknet-gauntlet-ledger` module, maybe you meant some signer module that takes care of the wallet definition?
Anyways, it's a submodule of the `starknet-gauntlet` atm, looks logical to me as it is.

Reply:
So this is what I meant:

- `gauntlet-starknet` should know nothing about specific Ledger/Trezor/Vault/etc. wallets.
- `gauntlet-starknet` should define a Wallet API (interfaces, types required by the framework) and it can provide a default implementation (random or/and read-pk-from-env, for example)
-  Support for all other wallets should be external to `gauntlet-starknet`, they should be implemented in some other package and hooked in by the user when setting up their Gauntlet project.
- To achieve this here, I suggested to (1) remove all ledger specific code from `gauntlet-starknet` package, (2) move the code to a new `gauntlet-starknet-ledger` package, (3) hook in the Ledger wallet in the `gauntlet-starknet-cli` package (final project).